pkg/authorize/ssl/ssl.go

```diff
@@ -52,7 +63,57 @@ func ClientCertInfoAsHeaders(config ClientCertConfig, logger log.Logger) func(ht
 				http.Error(w, "invalid secret", http.StatusForbidden)
 				return
 			}
-			next.ServeHTTP(w, r)
+
+			cnInfo := r.Header.Get(config.Config.CommonNameHeader)
+			if cnInfo == "" {
+				level.Info(logger).Log("msg", "invalid format for organisation and CN", "cnInfo", cnInfo)
+				http.Error(w, fmt.Sprintf("subject is empty. organisation and common name must be sent in request header %s",
+					config.Config.CommonNameHeader), http.StatusForbidden)
+				return
+			}
+
+			// cnInfo must be provided in the format of "/O = xyz, /CN = 123
+			parts := strings.Split(cnInfo, ",")
+			if len(parts) != 2 {
+				level.Info(logger).Log("msg", "invalid format for organisation and CN", "cnInfo", cnInfo)
+				http.Error(w, fmt.Sprintf("invalid format for organisation and CN %s",
+					config.Config.CommonNameHeader), http.StatusForbidden)
+				return
+			}
+
+			ctx := r.Context()
+			for i, part := range parts {
+				subPart := strings.Split(part, "=")
```

Comment:
Why aren't you using a regex for all these checks? Something like `/O = (?P<Org>[^,]+), /CN = (?P<CN>[^,]+)`

Reply:
for this I'd be inclined to just push out what we have and tidy up in a follow up if that is ok? I still haven't been able to see an actual request as routed through the proxy so the format is still unclear to me as of now